packages/gamut/src/Form/Input.tsx

```diff
@@ -1,32 +1,94 @@
-import cx from 'classnames';
-import React, { forwardRef, InputHTMLAttributes } from 'react';
+import { AlertIcon, CheckCircledIcon } from '@codecademy/gamut-icons';
+import { css } from '@emotion/react';
+import styled, { StyledComponent } from '@emotion/styled';
+import React, { forwardRef, InputHTMLAttributes, useState } from 'react';
 
-import styles from './styles/Input.module.scss';
+import { Box } from '../Box';
+import {
+  conditionalStyleProps,
+  conditionalStyles,
+  formBaseFieldStyles,
+  formFieldStyles,
+  iconStyles,
+} from './styles/shared';
 
 export type InputProps = InputHTMLAttributes<HTMLInputElement> & {
+  id?: string;
   className?: string;
+  activated?: boolean;
   error?: boolean;
   htmlFor?: string;
+  label?: string;
   name?: string;
   placeholder?: string;
   required?: boolean;
   type?: string;
+  valid?: boolean;
 };
 
-export const Input = forwardRef<HTMLInputElement, InputProps>(
-  ({ error, htmlFor, className, id, ...rest }, ref) => {
-    const classNames = cx(
-      styles.Input,
-      {
-        [styles.fileInput]: rest.type === 'file',
-      },
-      {
-        [styles.error]: error,
-      },
-      className
-    );
+export type InputWrapperProps = InputProps & {
```

Comment:
Should this omit `activated`? since it is only set in local state, and can't be overridden?
```suggestion
export type InputWrapperProps =Omit<InputProps, 'activated'> & {
```

Reply:
actually, you might want `InputProps` to exclude activated, since that's what folks will import, and include it in the type you use when you pass props to `InputElement`

Reply:
definitely - fixed!